src/viur/core/bones/key.py

```diff
@@ -76,81 +75,38 @@ def singleValueFromClient(self, value, skel, bone_name, client_data):
 
         return key, None
 
-    def unserialize(self, skel: 'viur.core.skeleton.SkeletonValues', name: str) -> bool:
-        """
-        This method is the inverse of :meth:serialize. It reads the key value from the datastore
-        and populates the corresponding KeyBone in the Skeleton. The method converts the value from
-        the datastore into an appropriate format for further use in the program.
-
-        :param skel: The SkeletonValues instance this bone is a part of.
-        :param name: The property name of this bone in the Skeleton (not the description).
-
-        :return: A boolean value indicating whether the operation was successful. Returns True if
-            the key value was successfully unserialized and added to the accessedValues of the
-            Skeleton, and False otherwise.
-
-        .. note:: The method contains an inner function, fixVals(val), which normalizes and
-            validates the key values before populating the bone.
-        """
-
-        def fixVals(val):
-            if isinstance(val, str):
-                try:
-                    val = utils.normalizeKey(db.Key.from_legacy_urlsafe(val))
-                except:
-                    val = None
-            elif not isinstance(val, db.Key):
-                val = None
-            return val
-
-        if (name == "key"
-            and isinstance(skel.dbEntity, db.Entity)
-            and skel.dbEntity.key
-                and not skel.dbEntity.key.is_partial):
+    def singleValueUnserialize(self, val):
+        if isinstance(val, db.Key):
+            rval = db.normalizeKey(val)
+        else:
+            rval, err = self.singleValueFromClient(val)
```

Comment:
I don't think it's a good idea to use `singleValueFromClient` here. There are the `allowed_kind`, existance `check` and `isInvalid` (`vFunc`) checks. I on of these check logic has changed or something that causes the saved key to be okay before, but not anymore the unserialize will now crash.

Reply:
This guarantees that bones that previously where e.g. StringBone or RawBone are being correctly converted to KeyBones. Can you give an example for your considerations?

Reply:
Image you have a `TreeSkel`, where keybone was initially simply 

```
parententry = KeyBone(
    descr="Parent",
)
```

but than you decided nesting of entries should be limited. So you implement an `isInvalid` method (not tested):

```
def is_depth_okay(value: db.Key) -> str | None:
    """parent is okay, if it's depth is max 3"""
    depth = 1
    while True:
        skel = self.viewSkel("node")
        skel.fromDB(value)
        if skel["parententry"] == skel["parentrepo"]:  # We reached the top level
            break
        depth += 1
        value = skel["parententry"]
    if depth > 3:
        return f"depth of {depth} is too much"
    return None
```


and use it for your bone:
```
parententry = KeyBone(
    descr="Parent",
    vfunc=is_depth_okay,
)
```


I there are now nested structures, which had already more than the 3 allowed parents (bc before this was not forbidden), the key would be illegal to use.
If u use now `singleValueFromClient` (where the `isInvalid` check is performed) in `singleValueUnserialize`, this would refuse to load the value. 

It is therefore no longer even possible to load the entity into a skeleton in order to change it with the new rules. in unserialise, only the format of the entry should be changed from database to skeleton format, but no ‘old’ data should be validated as invalid.

Reply:
Hallo @sveneberth,

I've modified the call to `singleValueFromClient` to avoid calling the `isInvalid()` function. Does this resolve your considerations?

Reply:
@sveneberth still awaiting your response.

Reply:
> @sveneberth still awaiting your response.

@sveneberth is still waiting for https://github.com/viur-framework/vi-vue-utils/issues/13 to test this. But so far this looks okay, and there's no failing unittest, so it must be okay ^^